lib/archethic/account/mem_tables/uco_ledger.ex

```diff
@@ -61,13 +63,12 @@ defmodule Archethic.Account.MemTables.UCOLedger do
       }
 
   """
-  @spec add_unspent_output(binary(), UnspentOutput.t(), DateTime.t()) :: :ok
+  @spec add_unspent_output(binary(), UnspentOutput.t()) :: :ok
   def add_unspent_output(
         to,
-        %UnspentOutput{from: from, amount: amount, reward?: reward?},
-        timestamp = %DateTime{}
+        %UnspentOutput{from: from, amount: amount, reward?: reward?, timestamp: timestamp}
       )
-      when is_binary(to) and is_integer(amount) and amount > 0 do
+      when is_binary(to) and is_integer(amount) and amount > 0 and not is_nil(timestamp) do
```

Comment:
I don't think you need to check timestamp as not nil, as it should always be a DateTime struct.